src/Renci.SshNet/Security/Cryptography/Ciphers/AesGcmCipher.cs

```diff
@@ -12,10 +10,15 @@ namespace Renci.SshNet.Security.Cryptography.Ciphers
     /// AES GCM cipher implementation.
     /// <see href="https://datatracker.ietf.org/doc/html/rfc5647"/>.
     /// </summary>
-    internal sealed class AesGcmCipher : SymmetricCipher, IDisposable
+    internal sealed partial class AesGcmCipher : SymmetricCipher, IDisposable
     {
+        private const int PacketLengthFieldLength = 4;
         private readonly byte[] _iv;
-        private readonly AesGcm _aesGcm;
+#if NET6_0_OR_GREATER
+        private readonly Impl _impl;
+#else
+        private readonly BouncyCastleImpl _impl;
+#endif
```

Comment:
nit, looks like it can be simplified

```suggestion
        private readonly Impl _impl;
```

Reply:
Code analysis doesn't like your suggestion

Reply:
See https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1859

Reply:
Ok, as the classes are sealed then there is unlikely to be a difference in performance, but there's no need to fight the code analysis